### PR TITLE
Anthro collection object module

### LINKED ISSUE #157: Converter module for Cataloging Anthro
At this time, this work will include support for fields included in the OHC remains collectionobject data in the migration queue that:

- do not exist in collectionobjects_common; 
- or are in collectionobjects_common but are overridden in collectionobjects_anthro 

This module will initially include the listed fields in the following namespaces, to keep things simple (and not have to create separate modules for collectionobjects_annotation and collectionobjects_nagpra: 

- ns2:collectionobjects_anthro
  - ageRange
  - behensmeyerSingleLower
  - behensmeyerUpper
  - commingledRemainsNote
  - fieldLocCounty
  - fieldLocPlace
  - fieldLocState
  - localityNote
  - minIndividuals
  - mortuaryTreatment
  - mortuaryTreatmentNote
- ns2:collectionobjects_annotation
  - annotationNote
  - annotationType
- ns2:collectionobjects_nagpra
  - nagpraReportFiledDate/dateDisplayDate
  - nagpraCategory
  - nagpraReportFiled
  - nagpraReportFiledBy
  - repatriationNote

This module will handle overrides to the following fields:
- objectProductionPeople (sources: concept/archculture,concept/ethculture)
- objectProductionPeopleRole (source: vocabularies:name(prodpeoplerole))

### LINKED ISSUE #169: Fix dependence on non-required field values for iteratively populating fieldGroups
For instance: https://github.com/collectionspace/cspace-converter/blob/a833bbb26831d615c87c764615e3350c3a0a1533/lib/collectionspace/converter/anthro/osteology.rb#L64-L78

`osteoAgeEstimateVerbatim` is not a required field. Other fields in the `osteoAgeEstimate` field may be present and need to be set when `osteoAgeEstimateVerbatim` has not been populated, or that column has been removed from the data being imported because the client doesn't need it. 

This code fails to import any data in this group if there is no data in for `osteoAgeEstimateVerbatim`

This problem exists throughout the cspace-converter code, and it should be fixed across the board. 

However, it's this part of the Osteology procedure module that is causing a problem importing client data now. It should be fixed first. (Mark cannot review/merge it until January 2 or after)

### PR DESCRIPTION
**Pull request by kspurgin**
This also fixes #169 for missing osteoAgeEstimateVerbatim values, which unblocks @meganforbes from loading Osteo procedure data. Ideally we will use the new, safer pattern everywhere else the code currently iterates on a non-required field to build a group.

It also adds a number of tests, including testing the publicart collectionobject. 

It adds a number of convenience methods and updates how add_group_lists works to make it clearer and able to support more variations.

### CODE REVIEW

File: spec/collectionspace/converter/core/collectionobject_spec.rb [hunk 1]
```diff
@@ -9,9 +9,10 @@
     '/document/*/objectNumber',
     '/document/*/numberOfObjects',
     '/document/*/titleGroupList/titleGroup/title',
-    { xpath: '/document/*/titleGroupList/titleGroup/titleLanguage',  transform: ->(text) { CSURN.parse(text)[:label] } },
+    { xpath: '/document/*/titleGroupList/titleGroup/titleLanguage',  transform: ->(text) { text.gsub!(/urn:.*?item:name\([^)]+\)'([^']+)'/, '\1') } },
```
Comment (kspurgin) [inline spec/collectionspace/converter/core/collectionobject_spec.rb]: @mark-cooper Is there a better way to make the tests pass when multiple URN values get returned? It seems that using `CSURN.parse(text)[:label]` in this proc only works if there's a single XML element returned. What I've got here works, but seems pretty clunky. I'm not real familiar with working with procs like this.

Comment (mark-cooper) [reply spec/collectionspace/converter/core/collectionobject_spec.rb]: @kspurgin you can use an index for the xpath to test one element (or more) specifically, for example:

https://github.com/collectionspace/cspace-converter/blob/master/spec/collectionspace/converter/anthro/osteology_spec.rb#L26-L27

For xpath first position is 1 not 0.

Comment (kspurgin) [reply spec/collectionspace/converter/core/collectionobject_spec.rb]: Thanks, @mark-cooper ! Now get back to being on vacation. :-D

Comment (kspurgin) [reply spec/collectionspace/converter/core/collectionobject_spec.rb]: My comments or mentions are just to make sure I don't forget them and do not indicate the need for any immediate attention :-)

File: lib/collectionspace/tools/xml.rb [hunk 2]
```diff
@@ -106,7 +272,93 @@ def self.process_array(xml, label, array)
...
+              if authority_config.keys.include?(field)
+                authority_type = authority_config[field]['authority'][0]
+                authority_name = authority_config[field]['authority'][1]
+                group[field] = Helpers.get_authority(authority_type, authority_name, value)
+              end
+            }
+          }
+        end
+
+        def self.to_boolean(value)
```
Comment (kspurgin) [inline lib/collectionspace/tools/xml.rb]: I could not figure out how to test CSRD.to_boolean and it didn't do what I needed to, so I created this to keep making progress. It seems like a lot of intermediate data wrangling stuff not directly to the XML was already in here when I started, and I've added a bunch more. Happy to move it elsewhere if that makes more sense.

**Comment by kspurgin** [comment]
@mark-cooper I think this is done and ready for review.

**Comment by mark-cooper** [comment]
@kspurgin nice (and a lot of) work! I like all the decisions you've made, and the inline method documentation was definitely overdue so thanks for getting the ball rolling on that ⭐️ 

I agree there's some refactoring that could be done around the XML and Helper classes (re: earlier comment), but nothing particularly urgent at this point.

**Pull request merged**